### ISSUE TITLE
API console for WFS and WMS?

### ISSUE THREAD

**Issue by BK01** [bcgov/DBC-APIM#93]
Investigate creating an OpenAPI spec for public DataBC WFS and WMS

Metadata record to have an API console:
https://catalogue.data.gov.bc.ca/dataset/bc-web-map-library

Docs:
https://medium.com/@cholmes/wfs-3-0-get-excited-yes-8e904fdbcc0
https://github.com/opengeospatial/WFS_FES/blob/master/overview.md

from @webgismd,
need sth like https://www.npmjs.com/package/wfs2swagger to produce Open API specs for our current WFS/WMS,etc

maybe align with incoming WFS3 ?
https://github.com/opengeospatial/WFS_FES
https://rawgit.com/opengeospatial/WFS_FES/master/docs/17-069.html

**Comment by BK01** [comment]
Discussing with web mapping and prepping for next week.

**Comment by banders** [comment]
@BK01 @ll911 Thinking more about this, I can't imagine any way to convert the WMS and WFS specs into OpenAPI format in a way that would be useful.  WMS and WFS are web services, but they are not RESTful.  (OAS is designed for RESTful APIs.)  Both WMS and WFS have a single endpoint with oodles of different parameter combinations.  OAS doesn't provide any means to organize and simplify this complexity.

Also, I tested the "wfs2swagger" tool to see if how they do the translation.  The tool doesn't work. It generates invalid OAS specs, and the specs include API endpoints that aren't valid WFS.

In short, I don't think it makes sense to create an OAS spec for WFS and WMS.

**Comment by BK01** [comment]
To revisit when WFS 3 is released which will be a RESTful spec.

**Comment by BK01** [comment]
@webgismd Temporarily on hold until WFS 3 API available for testing.
